DFC.Digital/DFC.Digital.Web.Sitefinity.CourseModule/Mvc/Controllers/CourseDetailsController.cs

```diff
@@ -14,27 +16,61 @@ public class CourseDetailsController : BaseDfcController
         #region Private Fields
         private readonly ICourseSearchService courseSearchService;
         private readonly IAsyncHelper asyncHelper;
+
         #endregion
 
         #region Ctor
-        public CourseDetailsController(IApplicationLogger loggingService, ICourseSearchService courseSearchService, IAsyncHelper asyncHelper) : base(loggingService)
+        public CourseDetailsController(IWebAppContext webAppContext, IApplicationLogger loggingService, ICourseSearchService courseSearchService, IAsyncHelper asyncHelper) : base(loggingService)
         {
             this.courseSearchService = courseSearchService;
             this.asyncHelper = asyncHelper;
+            this.WebAppContext = webAppContext;
         }
         #endregion
 
         #region Public Properties
 
         [DisplayName("Find a course Page")]
-        public string FindAcoursePage { get; set; } = "/find-a-course";
+        public string FindAcoursePage { get; set; } = "/course-directory/home/";
+
+        [DisplayName("No Course Description Message")]
+        public string NoCourseDescriptionMessage { get; set; } = "No course description available";
+
+        [DisplayName("No Entry Requirements Available Message")]
+        public string NoEntryRequirementsAvailableMessage { get; set; } = "No entry requirements available message";
+
+        [DisplayName("No Equipment Required Message")]
+        public string NoEquipmentRequiredMessage { get; set; } = "No equipment required";
+
+        [DisplayName("No Assessment Method Available Message")]
+        public string NoAssessmentMethodAvailableMessage { get; set; } = "Not available";
+
+        [DisplayName("No Venue Available Message")]
+        public string NoVenueAvailableMessage { get; set; } = "No venue Available";
+
+        [DisplayName("No Other Date Or Venue Available Message")]
+        public string NoOtherDateOrVenueAvailableMessage { get; set; } = "No other date or venue available";
+
+        protected IWebAppContext WebAppContext { get; set; }
 
         public ActionResult Index(string courseId)
         {
-            var viewModel = new CourseDetailsViewModel { FindACoursePage = FindAcoursePage };
+            var viewModel = new CourseDetailsViewModel
+            {
+                FindACoursePage = FindAcoursePage,
+            };
+
+            var referralUrl = HttpUtility.HtmlEncode(WebAppContext.RequestQueryString["referralUrl"]);
             if (!string.IsNullOrWhiteSpace(courseId))
             {
-                viewModel.CourseDetails = asyncHelper.Synchronise(() => courseSearchService.GetCourseDetailsAsync(courseId));
+                viewModel.CourseDetails = asyncHelper.Synchronise(() => courseSearchService.GetCourseDetailsAsync(courseId, referralUrl));
```

Comment:
If we are running sync, pls only use asyncHelper wrapper for async only if the underlying method has to be async. If it can be implmented sync all the way through that would be better than using the wrapper.
So does the underlying method has to be async?